akka-projection-kafka/src/main/scala/akka/projection/kafka/internal/KafkaSourceProviderImpl.scala

```diff
@@ -104,6 +107,13 @@ import org.apache.kafka.common.TopicPartition
         "The offset contains Kafka topic partitions that were revoked or lost in a previous rebalance")
   }
 
+  override def extractCreationTime(record: ConsumerRecord[K, V]): Long = {
```

Comment:
It should also be implemented by `EventSourcedProvider`. The timestamp is in the `EventEnvelope`.

Reply:
:+1: